docs/USAGE-GUIDE.md

```diff
@@ -1363,6 +1363,8 @@ ssize_t s2n_client_hello_get_extension_by_id(struct s2n_client_hello *ch, s2n_tl
 **s2n_client_hello_get_extension_length** returns the number of bytes the given extension type takes on the ClientHello message received by the server; it can be used to allocate the **out** buffer.
 **s2n_client_hello_get_extension_by_id** copies into the **out** buffer **max_length** bytes of a given extension type on the ClientHello and returns the number of copied bytes.
 
+If the return value is zero and s2n_errno is S2N_ERR_T_OK, the extension is present and empty. If the return value is zero and s2n_errno is not S2N_ERR_T_OK, the extension is not present
```

Comment:
I wonder is it even possible to end up in the case 'extension is present and empty'? That seems like a error in s2n at the point

```suggestion
**s2n_client_hello_get_extension_length** will return `0` if the extension is present but empty and also if the extension is not present. s2n_errno can be used to decipher between these two cases:

s2n_errno = 0;
ssize_t len = s2n_client_hello_get_extension_by_id(ch, 0);
if (len == 0 && s2n_errno == S2N_ERR_T_OK) {
        /** extension is present but empty **/
} else if (len == 0 && s2n_errno != S2N_ERR_T_OK) {
        /** extension is missing **/
} else {
    /** extension is preset with non-zero value **/
}
```

Reply:
hi @toidiu ,

> I wonder is it even possible to end up in the case 'extension is present and empty'?

It is possible for cases where the extension value is defined to be empty. For example for TLS session tickets, the client sends the extension with an empty value to signal to the server that it has no ticket and would like a new one.

will update the doc with your suggestion.